Repository/PhpcrRepository.php

```diff
@@ -119,4 +128,82 @@ protected function buildCollection(array $nodes)
 
         return $collection;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function add($path, $resource)
+    {
+        Assert::notEq('', trim($path, '/'), 'The root directory cannot be created.');
+        Assert::startsWith($path, '/', 'The target path %s is not absolute.');
+
+        $resolvedPath = $this->resolvePath($path);
+        $parentNode = $this->session->getNode(PathHelper::getParentPath($resolvedPath));
```

Comment:
I still think the injected path is the parentPath, so we don't have to extract it (same for odm). You can have a look into the filesyste implementation of puli itself?